### PR TITLE
Allow `single backticks` for inline monospaced text in docs

### LINKED ISSUE #3872: Can Doc literals use backticks for fixed-width spans?
I've been creating `Doc` literals for a year or so now and I still use a backtick (`` ` ``) instead of a double-single quote `''` _every_ single time. And looking through other docs, I often see other people make the same mistake. Can we use the standard markdown convention of backticks for fixed-width/code spans in `Doc` literals?

I believe that there is or was an issue with backticks being used for symbolic identifiers or something, but I couldn't easily find a historical discussion about this.

### PR DESCRIPTION
**Pull request by pchiusano**
This is a very small lexer change to allow a single <code>`</code> to start and end inline monospaced text in a doc, which people are very accustomed to from markdown. Tested in the doc parsing / pretty-printing transcript.

```
d = {{ This is `monospaced` text. }}
```

Previously, single backticks were used for infix function application for an alphanumeric operator, as in:

```haskell
foo `append` bar
```

However we long ago got rid of that syntax since the pretty-printer had no way to know when to use it. 

Thus, the single backtick syntax has been available for a while.

I left the pretty-printer alone in this PR, so it still prefers to use the `''` syntax when pretty-printing. Potentially we could have the pretty-printer use backticks preferentially, that'd be easy to do in a subsequent PR. Note that the `''` syntax is a bit more flexible re: escaping - you can start and end with any number of `''` (2 or more, and the end will require a matching number).

Thus, I see this PR as mostly being about avoiding accidental bad rendering from markdown muscle memory.

## Summarizing various doc syntax

```
''monospaced'' or `monospaced` or '''monospaced text'''
``1 + 1`` -- an inline code example (two backticks, it will be typechecked, and hyperlinked)
```

And then there's fenced code blocks, starting with at least three backticks, ending with the same number of backticks.

### CODE REVIEW

File: unison-src/transcripts-using-base/doc.output.md [hunk 1]
```diff
@@ -114,7 +114,7 @@ and the rendered output using `display`:
       section elements.
       
       Text can be **bold**, __italicized__, ~~strikethrough~~,
-      or ''monospaced''.
+      or ''monospaced'' (or ''monospaced'').
```
Comment (aryairani) [inline unison-src/transcripts-using-base/doc.output.md:117]: This is no big deal but also kind of weird. Do we actually want two syntaxes, and if so is this the one that we want shown after a round trip? Note that the rendered markdown below shows it as `` `monospaced` ``.

File: unison-src/transcripts/api-doc-rendering.output.md [hunk 2]
```diff
@@ -606,28 +604,11 @@ GET /api/getDefinition?names=term
...
-                                                        "tag": "Word"
-                                                    },
-                                                    {
-                                                        "contents": "`",
-                                                        "tag": "Word"
+                                                        "contents": {
+                                                            "contents": "\"doesn't typecheck\" + 1",
+                                                            "tag": "Word"
+                                                        },
+                                                        "tag": "Code"
```
Comment (aryairani) [inline unison-src/transcripts/api-doc-rendering.output.md:611]: is this change intended or ok? (The array length is smaller, with all words in a single element)

**Review by aryairani — APPROVED** [review]

**Comment by aryairani** [comment]
We might look at simplifying the Doc syntax later? Some of the markup still seems kind of weird to me.

File: unison-src/transcripts/api-doc-rendering.output.md (hunk 2 above)
Comment (pchiusano) [reply unison-src/transcripts/api-doc-rendering.output.md:611]: yeah, it's fine. There was a weird example in the api-docs, it looked like this:

```
`
"doesn't typecheck" + 1
`
```

with a single backtick on each line like that, which would previously parse as the literal backtick.
